pswalker/utils/argutils.py

```diff
@@ -72,3 +73,24 @@ def field_prepend(field, obj):
         field = obj.name
     
     return field
+
+
+def isiterable(obj):
+    """
+    Function that determines if an object is an iterable, not including 
+    str.
+
+    Parameters
+    ----------
+    obj : object
+        Object to test if it is an iterable.
+
+    Returns
+    -------
+    bool : bool
+        True if the obj is an iterable, False if not.
+    """
+    if isinstance(obj, str):
+        return False
+    else:
+        return isinstance(obj, Iterable)
```

Comment:
The only reason I've implemented this is because strings are considered iterables and instead of putting an explicit check for strings any time I've needed to, I just put it in the function instead. Also yeah I know, maybe we should talk to Bruce about a sensible way to pull it off, otherwise we will be stuck with just whatever AD has as PVs.